src/mame/video/bosco.cpp

```diff
@@ -300,10 +269,16 @@ WRITE_LINE_MEMBER(bosco_state::screen_vblank_bosco)
 	// falling edge
 	if (!state)
 	{
-		static const int speedsx[8] = { -1, -2, -3, 0, 3, 2, 1, 0 };
-		static const int speedsy[8] = { 0, -1, -2, -3, 0, 3, 2, 1 };
+		uint8_t speed_index_X, speed_index_Y;
+
+		// Bosconian scrolls in X and Y directions
+		speed_index_X = m_bosco_starcontrol[0] & 0x07;
+		speed_index_Y = (m_bosco_starcontrol[0] & 0x38) >> 3;
```

Comment:
Are these intermediate variables really needed?  If you want to keep them, at least make them `const`, like:
```
	const uint8_t speed_index_x = m_bosco_starcontrol[0] & 0x07;
	const uint8_t speed_index_y = (m_bosco_starcontrol[0] & 0x38) >> 3;
```
to make it clear they're just extracted fields and not intended to be changed later.

Reply:
Change committed